cmd/extensions/main.go

```diff
@@ -67,10 +67,13 @@ const (
 	defaultResync                = 30 * time.Second
 	apiServerSustainedQPSFlag    = "api-server-qps"
 	apiServerBurstQPSFlag        = "api-server-qps-burst"
+	sleepPeriod                  = "sleep-period"
+	sleepFailureThreshold        = "sleep-failure-threshold"
```

Comment:
Why two flags, when they're only ever used as `sleepPeriod * sleepFailureThreshold`.

I would suggest a single flag, `shutdownDurationFlag`, maybe call it `--shutdown-duration`. And make it a `pflag.Duration` - canonically anything that is a time duration should be a Duration.

Reply:
I named it ReadinessShutdownDuration just to clarify that its for the readiness check.